HEASARC XSPEC conda instalation and our test suite

For historical reasons, the CIAO conda installation of XSPEC (the model libraries) does not match the HEASARC approach. As there is now a HEASARC conda XSPEC installation, we actually hit this difference in two tests. We can easily fix the tests, but I wanted to have some documentation.

The test failures are not due to an issue in the code, bit are instead due to assumptions made in the test itself.

Here are the test failures:

```
=============================================================================== FAILURES ===============================================================================
______________________________________________________________________ test_manager_path_default _______________________________________________________________________

    @requires_xspec
    def test_manager_path_default():
        """Check the expected default setting for the manager path.
    
        Ideally this test would be run before any other
        tests of XSPEC are made (i.e. any XSPEC code is called).
        """
    
        # Is this always going to be correct?
        default_path = os.path.join(os.environ['HEADAS'],
                                    '../spectral/manager')
    
        from sherpa.astro import xspec
    
        oval = xspec.get_xspath_manager()
    
        # Normalize the paths to remove double / - e.g. if HEADAS is
        # set to /a/b/c/ rather than /a/b/c.
        #
>       assert os.path.normpath(oval) == os.path.normpath(default_path)
E       AssertionError: assert '/home/dburke...ctral/manager' == '/home/dburke...ctral/manager'
E         
E         - /home/dburke/micromamba/envs/xspec12150/spectral/manager
E         + /home/dburke/micromamba/envs/xspec12150/heasoft/spectral/manager
E         ?                                         ++++++++

sherpa/astro/xspec/tests/test_xspec_unit.py:249: AssertionError
_______________________________________________________________________ test_model_path_default ________________________________________________________________________

    @requires_xspec
    def test_model_path_default():
        """Check the expected default setting for the model data path.
    
        Ideally this test would be run before any other
        tests of XSPEC are made (i.e. any XSPEC code is called).
        """
    
        from sherpa.astro import xspec
    
        # Is this always going to be correct?
        #
        try:
            default_path = os.environ['XSPEC_MDATA_DIR']
        except KeyError:
            default_path = os.path.join(os.environ['HEADAS'],
                                        '../spectral/modelData/')
    
        oval = xspec.get_xspath_model()
    
        # Normalize the paths to remove double / - e.g. if HEADAS is
        # set to /a/b/c/ rather than /a/b/c.
        #
>       assert os.path.normpath(oval) == os.path.normpath(default_path)
E       AssertionError: assert '/home/dburke...ral/modelData' == '/home/dburke...ral/modelData'
E         
E         - /home/dburke/micromamba/envs/xspec12150/spectral/modelData
E         + /home/dburke/micromamba/envs/xspec12150/heasoft/spectral/modelData
E         ?                                         ++++++++

sherpa/astro/xspec/tests/test_xspec_unit.py:275: AssertionError
```

--- a/sherpa/astro/xspec/tests/test_xspec_unit.py
+++ b/sherpa/astro/xspec/tests/test_xspec_unit.py
@@ -235,18 +235,23 @@ def test_manager_path_default():
     tests of XSPEC are made (i.e. any XSPEC code is called).
     """
 
-    # Is this always going to be correct?
-    default_path = os.path.join(os.environ['HEADAS'],
-                                '../spectral/manager')
-
     from sherpa.astro import xspec
 
-    oval = xspec.get_xspath_manager()
+    oval = Path(xspec.get_xspath_manager()).resolve()
+    got = str(oval)
 
     # Normalize the paths to remove double / - e.g. if HEADAS is
     # set to /a/b/c/ rather than /a/b/c.
     #
-    assert os.path.normpath(oval) == os.path.normpath(default_path)
+    # The expected path depends on how XSPEC has been installed,
+    # since the CIAO-related environments do things a little
+    # differently to HEASARC ones, for historical reasons.
+    #
+    headas = Path(os.environ['HEADAS'])
+    heasarc_path = (headas / 'spectral/manager').resolve()
+    ciao_path = (headas / '../spectral/manager').resolve()
+
+    assert got in [str(heasarc_path), str(ciao_path)]
 
 
 @requires_xspec
@@ -259,20 +264,29 @@ def test_model_path_default():
 
     from sherpa.astro import xspec
 
-    # Is this always going to be correct?
-    #
-    try:
-        default_path = os.environ['XSPEC_MDATA_DIR']
-    except KeyError:
-        default_path = os.path.join(os.environ['HEADAS'],
-                                    '../spectral/modelData/')
-
-    oval = xspec.get_xspath_model()
-
     # Normalize the paths to remove double / - e.g. if HEADAS is
     # set to /a/b/c/ rather than /a/b/c.
     #
-    assert os.path.normpath(oval) == os.path.normpath(default_path)
+    oval = Path(xspec.get_xspath_model()).resolve()
+    got = str(oval)
+
+    # What is the best way to check the expected value?
+    #
+    # The expected path depends on how XSPEC has been installed,
+    # since the CIAO-related environments do things a little
+    # differently to HEASARC ones, for historical reasons.
+    #
+    key = 'XSPEC_MDATA_DIR'
+    if key in os.environ:
+        default_path = Path(os.environ[key]).resolve()
+        assert got == str(default_path)
+        return
+
+    headas = Path(os.environ['HEADAS'])
+    heasarc_path = (headas / 'spectral/modelData').resolve()
+    ciao_path = (headas / '../spectral/modelData').resolve()
+
+    assert got in [str(heasarc_path), str(ciao_path)]
 
 
 @requires_xspec

